Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -92,6 +92,92 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
                 "analyzers\\RoslynAnalyzer11.dll");
         }
 
+        // Test verifies https://jira.sonarsource.com/browse/SFSRAP-32
+        [TestMethod]
+        public void RoslynPlugin_GenerateForDependencyAnalyzers_Succeeds()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            string dummyContentFile = TestUtils.CreateTextFile("dummy.txt", outputDir, "non-analyzer content file");
+
+            // Create a valid analyzer package
+            RoslynAnalyzer11.CSharpAnalyzer analyzer = new RoslynAnalyzer11.CSharpAnalyzer();
+
+            string fakeRemoteNuGetDir = TestUtils.CreateTestDirectory(this.TestContext, ".fakeRemoteNuGet");
+            IPackageManager fakeRemotePkgMgr = CreatePackageManager(fakeRemoteNuGetDir);
+            IPackage child1 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child1", "1.1.0", analyzer.GetType().Assembly.Location);
+            IPackage child2 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child2", "1.2.0", analyzer.GetType().Assembly.Location);
+            IPackage targetPkg = AddPackage(fakeRemotePkgMgr, "Empty.Parent", "1.0.0", dummyContentFile, child1, child2);
+
+            string localPackageDestination = TestUtils.CreateTestDirectory(this.TestContext, ".localpackages");
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(fakeRemotePkgMgr.LocalRepository, localPackageDestination, logger);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            ProcessedArgs args = new ProcessedArgs(targetPkg.Id, targetPkg.Version, "cs", null, false, 
+                true /* generate plugins for dependencies with analyzers*/, outputDir);
+            bool result = apg.Generate(args);
+
+            // Assert
+            Assert.IsTrue(result);
+            string[] jarFilePaths = AssertPluginJarsExist(outputDir, 2); // Expecting one plugin per dependency with analyzers
```

Comment:
Checking for the existence of a single jar was ok previously because there was only supposed to be one. It isn't the right way to check the results now that could be multiple jars.
Instead, the checks should be something like the following:
_CheckJarForPackage(child1);
CheckJarForPackage(child2);
AssertPluginJarCount(2);_

_CheckJarForPackage_ would calculate the expected jar file name, check it exists and then check the contents.